tests/phpunit/tests/user/queryCache.php

```diff
@@ -0,0 +1,778 @@
+		$users2        = wp_list_pluck( $query2->get_results(), 'ID' );
+		$users_total2  = $query2->get_total();
+		$queries_after = get_num_queries();
+
+		$this->assertNotSame( $queries_before, $queries_after, 'Assert that queries are run' );
+		$this->assertNotSame( $users_total1, $users_total2, 'Assert that totals do not match' );
+		$this->assertSameSets( $users1, $users2, 'Results of the query are expected to match.' );
+	}
+
+	/**
+	 * @ticket 40613
+	 * @covers ::query
+	 */
+	public function test_query_cache_results() {
+		$args = array(
+			'cache_results' => true,
+		);
+
+		$query1 = new WP_User_Query( $args );
+		$users1 = wp_list_pluck( $query1->get_results(), 'ID' );
+
+		$queries_before = get_num_queries();
+
+		$args = array(
+			'cache_results' => false,
+		);
+
+		$query2        = new WP_User_Query( $args );
+		$users2        = wp_list_pluck( $query2->get_results(), 'ID' );
+		$queries_after = get_num_queries();
+
+		$this->assertNotSame( $queries_before, $queries_after, 'Assert that queries are run' );
+		$this->assertSameSets( $users1, $users2, 'Results of the query are expected to match.' );
+	}
+
+	/**
+	 * @ticket 40613
+	 * @covers ::query
+	 * @expectedDeprecated WP_User_Query
+	 */
+	public function test_query_cache_who() {
+		$args = array(
+			'who'    => 'authors',
+			'fields' => array( 'ID' ),
+		);
+
+		$query1       = new WP_User_Query( $args );
+		$users1       = $query1->get_results();
+		$users_total1 = $query1->get_total();
+
+		$queries_before = get_num_queries();
+		$query2         = new WP_User_Query( $args );
+		$users2         = $query2->get_results();
+		$users_total2   = $query2->get_total();
+		$queries_after  = get_num_queries();
+
+		$this->assertSame( $queries_before, $queries_after, 'No queries are expected run.' );
+		$this->assertSame( $users_total1, $users_total2, 'Number of users returned us expected to match.' );
+		$this->assertSameSets( $users1, $users2, 'Results of the query are expected to match.' );
+	}
+
+	/**
+	 * @ticket 40613
+	 * @covers ::query
+	 * @dataProvider data_returning_field_subset_as_array
+	 * @param array $args Optional. See WP_User_Query::prepare_query()
+	 */
+	public function test_query_cache( array $args ) {
+		$query1       = new WP_User_Query( $args );
+		$users1       = $query1->get_results();
+		$users_total1 = $query1->get_total();
+
+		$queries_before = get_num_queries();
+		$query2         = new WP_User_Query( $args );
+		$users2         = $query2->get_results();
+		$users_total2   = $query2->get_total();
+		$queries_after  = get_num_queries();
+
+		$this->assertSame( 0, $queries_after - $queries_before, 'Assert that no queries are run' );
+		$this->assertSame( $users_total1, $users_total2, 'Assert that totals do match' );
+		$this->assertSameSets( $users1, $users2, 'Asset that results of query match' );
+	}
+
+	/**
+	 * Data provider
+	 *
+	 * @return array
+	 */
+	public function data_returning_field_subset_as_array() {
```

Comment:
```suggestion
	public function data_query_cache() {
```

When a data provider is only used by one test, convention is to give it the same name.

Reply:
Fixed https://github.com/WordPress/wordpress-develop/pull/3544/commits/61b74c64deb5e060c9866500499272c0dbb2a10d